gammapy/modeling/fit.py

```diff
@@ -721,6 +725,75 @@ def covariance_result(self):
         """Optimize result."""
         return self._covariance_result
 
+    def to_dict(self, full_output=False, overwrite_templates=False):
+        """Convert to dictionary."""
+
+        models_dict = self.models.to_dict(
+            full_output=full_output, overwrite_templates=overwrite_templates
+        )
+
+        return {
```

Comment:
Maybe start with `optimize_results` then `covariance_result` and finally `models`.